File: tests/XMP/global-view/structure/C/loop.c
#include <stdio.h>
#include <stdlib.h>
#include <xmp.h>
#pragma xmp nodes p[*]
#pragma xmp template t[10]
#pragma xmp distribute t[block] onto p
struct b{
  int a[10];
};
#pragma xmp align b.a[i] with t[i]
struct b c;

int main(){
  int sum = 0;
#pragma xmp loop on t[i]
  for(int i=0;i<10;i++)
    c.a[i] = i;

#pragma xmp loop on t[i] reduction(+:sum)
  for(int i=0;i<10;i++)
    sum += c.a[i];

  if(sum != 45){
    if(xmpc_node_num() == 0) printf("Error !\n");
    exit(1);
  }
  else
    if(xmpc_node_num() == 0) printf("OK\n");
  
  return 0;
}

File: tests/XMP/global-view/structure/F/loop.f90
program	loop
  implicit none
  type b
     real(8):: a(10)
  end type b
!$xmp nodes p(*)
!$xmp template t(10)
!$xmp distribute t(block) onto p
!$xmp align b%a(i) with t(i)
  type(b):: c
  integer :: i, sum = 0

!$xmp loop on t(i)
  do i=1, 10
     c%a(i) = i
  end do

!$xmp loop on t(i) reduction(+:sum)
  do i=1, 10
     sum = sum + c%a(i)
  end do
  
  if (sum == 55) then
     if (xmp_node_num() == 1) then
        write(*,*) "OK"
     end if
  else
     if (xmp_node_num() == 1) then
        write(*,*) "Error!"
     end if
     call exit(1)
  end if
end program loop

  
